include/aws/sdkutils/resource_name.h

```diff
@@ -0,0 +1,44 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+#ifndef AWS_COMMON_RESOURCE_NAME_H
+#define AWS_COMMON_RESOURCE_NAME_H
+#pragma once
```

Comment:
trivial: AWS_SDKUTILS_RESOURCE_NAME_H

also, I know pragma once works everywhere, but we haven't been using it in our C code